lib/gambit/list/list.scm

```diff
@@ -607,22 +607,34 @@
                 (macro-check-list x 2 (assoc obj lst c)
                   #f))))))))
 
-(define-prim (##remq elem lst)
-
-  (include "~~lib/gambit/prim/prim#.scm") ;; map fx+ to ##fx+, etc
-
-  (define (rem elem lst)
-    (if (pair? lst)
-        (let* ((lst2 (rem elem (cdr lst)))
-               (x (car lst)))
-          (if (eq? x elem)
-              lst2
-              (if (eq? lst2 (cdr lst))
-                  lst
-                  (cons x lst2))))
-        '()))
-
-  (rem elem lst))
+(define-macro (##macro-filter prim-call list var test-expr)
```

Comment:
The name of the macro should be `macro-filter` without the `##`.

Reply:
OK, changed.